seed/utils/filter_state.py

```diff
@@ -0,0 +1,175 @@
+        cycle = Cycle.objects.filter(organization_id=org_id).order_by('name')
+        if cycle:
+            cycle = cycle.first()
+        else:
+            return JsonResponse({
+                'status': 'error',
+                'message': 'Could not locate cycle',
+                'pagination': {
+                    'total': 0
+                },
+                'cycle_id': None,
+                'results': []
+            })
+
+    if state_type == 'property':
+        views_list = (
+            PropertyView.objects.select_related('property', 'state', 'cycle')
+            .filter(property__organization_id=org_id, cycle=cycle)
+        )
+    elif state_type == 'taxlot':
+        views_list = (
+            TaxLotView.objects.select_related('taxlot', 'state', 'cycle')
+            .filter(taxlot__organization_id=org_id, cycle=cycle)
+        )
+
+    include_related = (
+        str(request.query_params.get('include_related', 'true')).lower() == 'true'
+    )
+
+    # Retrieve all the columns that are in the db for this organization
+    columns_from_database = Column.retrieve_all(
+        org_id=org_id,
+        inventory_type=state_type,
+        only_used=False,
+        include_related=include_related
+    )
+    try:
+        filters, annotations, order_by = build_view_filters_and_sorts(request.query_params, columns_from_database)
+    except FilterException as e:
+        return JsonResponse(
+            {
+                'status': 'error',
+                'message': f'Error filtering: {str(e)}'
+            },
+            status=status.HTTP_400_BAD_REQUEST
+        )
+
+    views_list = views_list.annotate(**annotations).filter(filters).order_by(*order_by)
+
+    # Return property views limited to the 'property_view_ids' list. Otherwise, if selected is empty, return all
+    if f'{state_type}_view_ids' in request.data and request.data[f'{state_type}_view_ids']:
+        views_list = views_list.filter(id__in=request.data[f'{state_type}_view_ids'])
+
+    paginator = Paginator(views_list, per_page)
+
+    try:
+        views = paginator.page(page)
+        page = int(page)
+    except PageNotAnInteger:
+        views = paginator.page(1)
+        page = 1
+    except EmptyPage:
+        views = paginator.page(paginator.num_pages)
+        page = paginator.num_pages
+    except DataError as e:
+        return JsonResponse(
+            {
+                'status': 'error',
+                'recommended_action': 'update_column_settings',
+                'message': f'Error filtering - your data might not match the column settings data type: {str(e)}'
+            },
+            status=status.HTTP_400_BAD_REQUEST
+        )
+
+    # This uses an old method of returning the show_columns. There is a new method that
+    # is prefered in v2.1 API with the ProfileIdMixin.
+    if state_type == 'property':
+        view_list_state = VIEW_LIST_PROPERTY
+    elif state_type == 'taxlot':
+        view_list_state = VIEW_LIST_TAXLOT
```

Comment:
```suggestion
    if state_type == 'property':
        profile_inventory_type = VIEW_LIST_PROPERTY
    elif state_type == 'taxlot':
        profile_inventory_type = VIEW_LIST_TAXLOT
```

I think this might be a bit more clear about what it is (an enum for ColumProfile.inventory_type)